librz/asm/arch/evm/evmasm.c

```diff
@@ -0,0 +1,359 @@
+// SPDX-FileCopyrightText: 2023 gogo <gogo246475@gmail.com>
+// SPDX-FileCopyrightText: 2012-2020 pancake <pancake@nopcode.org>
+// SPDX-License-Identifier: LGPL-3.0-only
+
+#include <rz_util.h>
+#include <rz_types.h>
+#include <rz_asm.h>
+#include <string.h>
+#include <stdlib.h>
+
+static int evmAsm(RzAsm *a, RzAsmOp *op, const char *buf) {
```

Comment:
You always assume that `buf` contains the correct information, but what happens if its length is less than expected? based on your code, many out of bound reads. you should always check for boundaries.